# Homework 4 - Data Structures, Conditional Statements, Loops
## Mariah N Cornelio

___

## Data Structures
### 1. List and Tuple:

- Create a list named fruits containing the following: "apple", "banana", "cherry".
- Convert this list into a tuple.
- Add "orange" to the list of fruits. 
- Show the difference in the content of the list and tuple after the addition. 

In [1]:
fruits=["apple", "banana", "cherry"]

In [2]:
tuple(fruits)

('apple', 'banana', 'cherry')

In [3]:
fruits.append("orange")

In [4]:
tuple(fruits)

('apple', 'banana', 'cherry', 'orange')

In [5]:
list(fruits)

['apple', 'banana', 'cherry', 'orange']

### 2. Dictionary

- Create a dictionary named person with keys "first_name", "last_name", and "age". Populate it with values of your choice.
- Add a new-value pair, "city" with a value of your choice.
- Display the keys in the dictionary using a Python command.

In [6]:
person={"first_name":"Meowmeow", "last_name":"Farts", "age":"21"}

In [7]:
person["city"]="Dallas"

In [8]:
person.keys()

dict_keys(['first_name', 'last_name', 'age', 'city'])

### 3. Set

- Create a set named colors with the values "red", "blue", "green".
- Add "yellow" to the set and remove "blue".
- Check if "pink" is in the set and display the result.

In [9]:
colors={"red", "blue", "green"}

In [10]:
colors.add("yellow")
colors.remove("blue")

In [11]:
colors

{'green', 'red', 'yellow'}

In [12]:
if "pink" in colors:
    print("True")
else:
    print("False")

False


---

## Conditional Statements
### 4. Write a Python program that takes an input from the user and checks if the number is positive, negative, or zero.

In [13]:
number=int(input("Enter in a number: "))

if number > 0:
    print("This number is positive.")
elif number < 0:
    print("This number is negative.")
elif number==0:
    print("This number is 0.")

Enter in a number: -6
This number is negative.


### 5. Given a list of numbers: [23,45,8,12,78,56,89], write a Python program that prints out all even numbers from the list using a conditional statement.

In [14]:
numlist=[23,45,8,12,78,56,89]

for n in numlist:
    if n%2==0:
        print("This number in the list is even: ", n)

This number in the list is even:  8
This number in the list is even:  12
This number in the list is even:  78
This number in the list is even:  56


### 6. Write a program that asks the user to enter their age. Based on the user's input, determine if they are eligible to vote (consider the voting age as 18). 

In [15]:
age = int(input("What is your age? "))

if age >= 18:
    print("Congratulations! You are able to vote.")
else:
    print("You are not eligible to vote.")

What is your age? 19
Congratulations! You are able to vote.


---

## While Loop:
### 7. Write a program that prompts the user to enter numbers continuously (using a while loop) and stops taking inputs when the user enters "q". After that, display the sum of all the numbers created.

In [16]:
numbers=""
total=[]

while numbers!="q":
    numbers=input("Enter in a number. Type 'q' when ready to quit: ")
    total.append(numbers)

total.remove('q') # This removes the "q" string from the list
total
int_total = list(map(int,total)) # This turns the total list into integers
total_sum = sum(int_total)

print("The sum of all the numbers you entered is", total_sum)

Enter in a number. Type 'q' when ready to quit: 24
Enter in a number. Type 'q' when ready to quit: 68
Enter in a number. Type 'q' when ready to quit: 94
Enter in a number. Type 'q' when ready to quit: 2
Enter in a number. Type 'q' when ready to quit: q
The sum of all the numbers you entered is 188


In [17]:
# This took me so long to figure out...

## 8. Create a Python program. using a while loop that prints numbers from 10 down to 1 and then prints "LIFTOFFFFF!"

In [18]:
x = 10

print("Countdown commencing...")
while x>0:
    print(x)
    x=x-1
    
print("LIFTOFFF!!!!")

Countdown commencing...
10
9
8
7
6
5
4
3
2
1
LIFTOFFF!!!!


### 9. Create a program that asks the user for a number and then prints out the multiplication table for that number using a while loop.

In [19]:
numInput = int(input("Enter a # to get its multiplication table up to that number: "))
num = numInput

while num <= numInput:
    i = 1
    while i <= numInput:
        product = num*i
        print(num, " * ", i, " = ", product)
        i = i + 1
    print("") 
    break # this line of code stops the loop 

Enter a # to get its multiplication table up to that number: 12
12  *  1  =  12
12  *  2  =  24
12  *  3  =  36
12  *  4  =  48
12  *  5  =  60
12  *  6  =  72
12  *  7  =  84
12  *  8  =  96
12  *  9  =  108
12  *  10  =  120
12  *  11  =  132
12  *  12  =  144



---

## For Loop
### 10. Given a list: ["apple", "banana", "cherry", "date"], use a for loop to print each fruit in the list.

In [20]:
fruits = ["apple", "banana", "cherry", "date"]

for f in fruits:
    print(f)

apple
banana
cherry
date


### 11. Using a for loop, calculate the factorial of a given number (n!). For example, if the given number is 5, the output should be 5x4x3x2x1=120.

In [21]:
n=int(input("Enter in a number: "))
factorial = 1

if n>=1:
    for i in range(1,n+1):
        factorial=factorial*i
print("The factorial for this number is", factorial)   

Enter in a number: 6
The factorial for this number is 720


### 12. Create a program using a for loop that prompts the user to enter five numbers. After all the numbers are entered, the program should display the average of the numbers.

In [22]:
nums=list(map(int, input("Enter in 5 numbers with no spaces in between them: ").split()))

def Avg(nums): # Making an averaging function called (Avg)
    list_sum=0
    for i in range(len(nums)):
        list_sum += nums[i]
    average=list_sum/len(nums)
    return average

average=Avg(nums)
print("The average of the numbers you entered is", round(average, 2))

Enter in 5 numbers with no spaces in between them: 88 66 44 22 2
The average of the numbers you entered is 44.4


---